apps/modernization-ui/src/pages/addPatient/components/Race/RaceFields.tsx

```diff
@@ -2,30 +2,78 @@ import { Checkbox, Fieldset, Grid } from '@trussworks/react-uswds';
 import FormCard from '../../../../components/FormCard/FormCard';
 import { SearchCriteriaContext } from '../../../../providers/SearchCriteriaContext';
 import { formatInterfaceString } from '../../../../utils/util';
+import { Controller } from 'react-hook-form';
 
 export interface InputEthnicityFields {
     ethnicity: string;
     // TODO races
 }
-export default function RaceFields({ title, id }: { id?: string; title?: string }) {
+export default function EthnicityFields({ id, title, control }: { id?: string; title?: string; control?: any }) {
+    const tempArr: any = [];
     return (
         <FormCard title={title} id={id}>
             <Grid col={12} className="padding-x-3 padding-bottom-3">
                 <Grid row>
                     <Grid col={12}>
                         <Fieldset>
-                            <SearchCriteriaContext.Consumer>
-                                {({ searchCriteria }) =>
-                                    Object.values(searchCriteria.races).map((race, index) => (
-                                        <Checkbox
-                                            key={index}
-                                            id={race.id.code}
-                                            name={race.id.code}
-                                            label={formatInterfaceString(race.codeDescTxt)}
-                                        />
-                                    ))
-                                }
-                            </SearchCriteriaContext.Consumer>
+                            <Controller
+                                control={control}
+                                name="race"
+                                render={({ field: { onChange, value } }) => {
+                                    console.log('value:', value);
+                                    return (
+                                        <SearchCriteriaContext.Consumer>
+                                            {({ searchCriteria }) =>
+                                                Object.values(searchCriteria.races).map((race, index) => (
+                                                    <Checkbox
+                                                        key={index}
+                                                        onChange={(e) => {
+                                                            if (!value || value?.length === 0) {
+                                                                if (e.target.checked) {
+                                                                    tempArr.push(e.target.value);
+                                                                    console.log('true:', tempArr);
```

Comment:
do we want to keep these log statements in?

Reply:
Not needed, will remove